src/compiler/checker.ts

```diff
@@ -16325,12 +16325,39 @@ namespace ts {
             if (moduleSymbol) {
                 const esModuleSymbol = resolveESModuleSymbol(moduleSymbol, specifier, /*dontRecursivelyResolve*/ true);
                 if (esModuleSymbol) {
-                    return createPromiseReturnType(node, getTypeOfSymbol(esModuleSymbol));
+                    return createPromiseReturnType(node, syntheticDefaultType(getTypeOfSymbol(esModuleSymbol), esModuleSymbol));
                 }
             }
             return createPromiseReturnType(node, anyType);
         }
 
+        function syntheticDefaultType(type: Type, symbol: Symbol): Type {
+            if (allowSyntheticDefaultImports && type && type !== unknownType) {
+                const synthType = type as SyntheticDefaultModuleType;
+                if (!synthType.syntheticType) {
+                    const defaultSymbol = getPropertyOfType(type, InternalSymbolName.Default);
+                    if (!defaultSymbol) {
+                        const memberTable = createSymbolTable();
+                        const newSymbol = createSymbol(SymbolFlags.Alias, InternalSymbolName.Default);
+                        getSymbolLinks(newSymbol).target = resolveSymbol(symbol);

```

Comment:
The `getSymbolLinks` call is unnecessary here. `newSymbol` is already a `TransientSymbol` which is its own `SymbolLinks`.